ladybug/stat.py

```diff
@@ -130,20 +161,87 @@ def import_data(self):
             self._location.time_zone = time_zone
             self._location.elevation = elevation
 
-            # defaults in case the climate is unclassifiable
-            self._ashrae_climate_zone = None
-            self._koppen_climate_zone = None
-
-            # move through the document and pull out the climate zone and tau values
+            # move through the document and pull out:
+            # The tau values for the sky model
+            # The monthly temperatures (for design days)
+            # The climate zone
+            winter_des_section = False
+            summer_des_section = False
+            wind_spd_section = False
+            wind_dir_section = -1
             for line in statwin:
                 if 'taub (beam)' in line:
-                    taub_raw = line.replace('taub (beam)', '').strip().split('\t')
-                    self._monthly_tau_beam = [float(i) if 'N' not in i
-                                              else None for i in taub_raw]
+                    self._monthly_tau_beam = self._split_monthly_row(
+                        line, 'taub (beam)')
                 elif 'taud (diffuse)' in line:
-                    taud_raw = line.replace('taud (diffuse)', '').strip().split('\t')
-                    self._monthly_tau_diffuse = [float(i) if 'N' not in i
-                                                 else None for i in taud_raw]
+                    self._monthly_tau_diffuse = self._split_monthly_row(
+                        line, 'taud (diffuse)')
+                elif 'Design Stat	Coldest' in line:
+                    winter_des_section = True
+                    winter_keys = self._split_des_day_row(
+                        line, 'Design Stat	Coldest')
+                elif winter_des_section is True and 'Heating' in line:
+                    winter_des_section = False
+                    winter_vals = self._split_des_day_row(
+                        line, 'Heating')
+                    self._winter_des_day_dict = {}
+                    for key, val in zip(winter_keys, winter_vals):
+                        self._winter_des_day_dict[key] = val
+                elif 'Design Stat	Hottest' in line:
```

Comment:
What about using regular expression to avoid all these if, elif statements?

Reply:
In the end, this turned out to be the same number of lines but I guess you are right that it's more readable.

Reply:
I'm also realizing that using regex here is making the code more prone to failure in the event that a given data type is not found in the .stat file.  Essentially, all of these if statements help guard in the event that there's missing data in the file.  If I want to make my regex code as resilient, I end up having to add back all of the if statements anyway to check and see if the regex successfully found a match.  Does this still sound worthwhile to you, @mostaphaRoudsari ?

Reply:
I implemented the regex as discussed